src/LazyFunction.h

```diff
@@ -0,0 +1,43 @@
+//
+// Created by matt on 12/09/2019.
+//
+
+#ifndef NOVELRT_LAZYFUNCTION_H
+#define NOVELRT_LAZYFUNCTION_H
+#include <functional>
+#include <memory>
+
+namespace NovelRT {
+template<typename T>
+class LazyFunction {
```

Comment:
just interested: Why `LazyFunction` as the name?

Reply:
I was tired and it seemed appropriate. Shrug. I can change it if you think it should just be Lazy or something else.

Reply:
I'm not sure its the most fitting since its a value that is lazily initialized and not necessarily a function that is lazy to call.
Maybe `LazyValue` if you didn't like just `Lazy`?

Reply:
I'll address this now.